regcore/models.py

```diff
@@ -5,6 +5,15 @@ class PartQuerySet(models.QuerySet):
     def effective(self, date):
         return self.filter(date__lte=date).order_by("name", "-date").distinct("name")
 
+    def effective_title(self, date, title):
+        return self.filter(title=title).filter(date__lte=date).order_by("name", "-date").distinct("name")
+
+    def full_part(self, date, title, part):
+        return self.filter(title=title).filter(date__lte=date).filter(name=part).latest("date")
+
+    def versions(self, title, part):
+        return self.filter(name=part).filter(title=title).order_by('-date')
```

Comment:
Can be more specific here (abstractions should always hide something). And fetch the value specifically so it's not pulling down everything. I think `values('date')` will work but I'm not sure.